src/index.spec.ts

```diff
@@ -179,6 +186,217 @@ describe('contracts', () => {
   });
 });
 
+describe('check errors', () => {
+  it('tuple type', () => {
+    try {
+      const Check = Tuple(Number, String, Boolean);
+      Check.check([false, '0', true]);
+      fail("tuple passed, eventhough should've failed");
+    } catch (e) {
+      expect(e.message).toBe('Expected a number, but was boolean');
+      expect(e.key).toBe('[0]');
+    }
```

Comment:
This same structure is repeated for all the error message checks, how about extracting a function to eliminate the repetition and make clear what in particular is being tested in each case?